.github/workflows/onPRReviews.yml

```diff
@@ -1,23 +1,119 @@
   pull_request_review:
     types: [edited, dismissed, submitted]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
 jobs:
-  checkMergeReady:
+  get-pr-number:
+    name: Get Current PR Number
+    runs-on: ubuntu-latest
+    outputs: 
+      pr-number: ${{ steps.condval.outputs.value }}
+    steps:
+      - uses: 3drepo/action-cond@v1
+        id: condval
+        with:
+          cond: ${{ github.event_name == 'pull_request' }}
+          if_true: ${{ github.event.number }}
+          if_false: ${{ github.event.pull_request.number }}
+
+  get-issue-number:
+    name: Get Current Issue Number
+    needs: [ get-pr-number ]
+    runs-on: ubuntu-latest
+    outputs: 
+      issue-number: ${{ steps.commitMsgParser.outputs.issue-number }}
+      content-id: ${{ steps.commitMsgParser.outputs.content-id }}
+    steps:
+      - name: Extract Issue Number from PR
+        uses: 3drepo/extract-issue-number-from-PR@v1
+        with:
+          pr: ${{ needs.get-pr-number.outputs.pr-number }}
+          base: staging
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+        id: commitMsgParser
+
+  check-merge-ready:
     name: Ensure all reviewers have approved the PR
     runs-on: ubuntu-latest
+    needs: [ get-issue-number ]
+    outputs: 
+      approved: ${{ steps.checkReviews.outputs.approved }}
     steps:
-      - uses: haya14busa/action-cond@v1
+      - uses: 3drepo/action-cond@v1
         id: condval
         with:
           cond: ${{ github.event_name == 'pull_request' }}
           if_true: ${{ github.event.number }}
           if_false: ${{ github.event.pull_request.number }}
       - name: Check reviews
-        uses: carmenfan/all-reviewers-approved@v1.0
+        id: checkReviews
+        uses: 3drepo/all-reviewers-approved@v1
         with:
           pr: ${{ steps.condval.outputs.value }}
           github-token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Check reviews 
+        if: failure()
+        id: setStatusOutputToTrue   
+        run: exit 0
 
+#  if someone has been assigned a reviewer -> change issue status to "awaiting approval"
+  update-project-issue-status:
+    name: Update Project Issue Status
+    if: ${{ always() && needs.check-merge-ready.outputs.approved = '' }} # if not reviewer approved let's modify the status.
+    runs-on: ubuntu-latest
+    outputs: 
+      content-id: ${{ needs.get-issue-number.outputs.content-id }}
+      status: ${{ steps.setStatusOutput.outputs.status }}
+    needs: [ get-issue-number, check-merge-ready ]
+    steps:
+      - name: Set status message from input 
+        id: setStatusOutput   
+        run: |
+          set -ex 
+          case $ACTION in
+            submitted ) 
+              echo "status=Address Feedback" >> $GITHUB_OUTPUT
+              ;;
+            review_requested ) 
+              echo "status=To Review" >> $GITHUB_OUTPUT
```

Comment:
This block of code is not very easy to read, but i'm guessing what you're saying here is set issue status to  `To Review` if the trigger was `review_requested`?

`To Review` should never be set on a issue (this is a PR status), so it should be `Awaiting Approval`.

Reply:
I misunderstood your previous note for setting to Review on the PR and did it on the issue, so if we want to modify both status then I need a second block of this for PR action and one for Issue action. Is that what you're after?

Reply:
it's suppose to be done in the issue, what you done is correct?

the PR status is done via some actions wired in the project already all you have to worry about is the issue ones.

to be clear:

when a review is requested/the PR is first created, the issue should be in `awaiting approval`
someone done a review and it was not approved, the issue should be in `addressing feedback`

Reply:
ah I guess if someone re-requested a review the PR needs to be set to `to review`, is that what you are trying to do here?

Reply:
Yes, that was the intention